classes/helper.php

```diff
@@ -371,4 +374,78 @@ public static function regular_expression_search(string $search, string $table,
         }
         return $results;
     }
+
+    /**
+     * Get column info from column name.
+     *
+     * @param string $table The table name.
+     * @param string $columnname The column name.
+     */
+    public static function get_column_info(string $table, string $columnname): database_column_info {
+        global $DB;
+
+        $columns = $DB->get_columns($table);
+        $column = null;
+        foreach ($columns as $col) {
+            if ($col->name == $columnname) {
+                $column = $col;
+                break;
+            }
+        }
+
+        if (is_null($column)) {
+            throw new moodle_exception(get_string('errorcolumnnotfound', 'tool_advancedreplace', $columnname));
+        }
+
+        return $column;
+    }
+
+    /**
+     * A clone of the core function replace_all_text.
+     * We have optional id parameter to restrict the search.
+     *
+     * @since Moodle 2.6.1
+     * @param string $table name of the table
+     * @param database_column_info $column
+     * @param string $search
+     * @param string $replace
+     * @param int $id optional id to restrict the search
+     */
+    public static function replace_all_text($table, database_column_info $column, $search, $replace, $id = null) {
+        global $DB;
+        if (!$DB->replace_all_text_supported()) {
+            return;
+        }
+
+        // Enclose the column name by the proper quotes if it's a reserved word.
+        $columnname = $DB->get_manager()->generator->getEncQuoted($column->name);
+
+        $searchsql = $DB->sql_like($columnname, '?');
+        $searchparam = '%'.$DB->sql_like_escape($search).'%';
+
+        // Add the id to the search.
+        if (!is_null($id)) {
+            $searchsql .= ' AND id = ?';
+            $params = [$search, $replace, $searchparam, $id];
+        } else {
+            $params = [$search, $replace, $searchparam];
+        }
+
+        $sql = "UPDATE {".$table."}
```

Comment:
can you move this down into the if, and then i the column type is not X or C then coding exception

Reply:
I changed this to swich case